File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/AgentSit.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;
import java.util.UUID;

/**
 * AgentSit - Actually sit on object
 *
 * <p>Template: {@code AgentSit High 7 NotTrusted Unencoded}
 * (recovered/reference/message_template.msg).
 */
public class AgentSit extends SLMessage {
    public AgentData AgentData_Field;

    /** Block AgentData, Single. */
    public static class AgentData {
        public UUID AgentID; // LLUUID
        public UUID SessionID; // LLUUID
    }

    public AgentSit() {
        this.zeroCoded = false;
        this.AgentData_Field = new AgentData();
    }

    @Override
    public int CalcPayloadSize() {
        return 33;
    }

    @Override
    public void Handle(SLMessageHandler messageHandler) {
        messageHandler.HandleAgentSit(this);
    }

    @Override
    public void PackPayload(ByteBuffer byteBuffer) {
        // Message number: High 7 (AgentSit).
        byteBuffer.put((byte) 0x07);
        packUUID(byteBuffer, this.AgentData_Field.AgentID);
        packUUID(byteBuffer, this.AgentData_Field.SessionID);
    }

    @Override
    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.AgentData_Field.AgentID = unpackUUID(byteBuffer);
        this.AgentData_Field.SessionID = unpackUUID(byteBuffer);
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/AgentSit.kt
package com.lumiyaviewer.lumiya.slproto.messages

import com.lumiyaviewer.lumiya.slproto.SLMessage
import java.nio.ByteBuffer
import java.util.UUID

/**
 * AgentSit - Actually sit on object
 *
 * <p>Template: {@code AgentSit High 7 NotTrusted Unencoded}
 * (recovered/reference/message_template.msg).
 */
open class AgentSit : SLMessage() {
    @JvmField var AgentData_Field: AgentData = AgentData()

    /** Block AgentData, Single. */
    open class AgentData {
        @JvmField var AgentID: UUID? = null
        @JvmField var SessionID: UUID? = null
    }

    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int {
        return 33
    }

    override fun Handle(messageHandler: SLMessageHandler) {
        messageHandler.HandleAgentSit(this)
    }

    override fun PackPayload(byteBuffer: ByteBuffer) {
        // Message number: High 7 (AgentSit).
        byteBuffer.put((0x07).toByte())
        packUUID(byteBuffer, AgentData_Field.AgentID)
        packUUID(byteBuffer, AgentData_Field.SessionID)
    }

    override fun UnpackPayload(byteBuffer: ByteBuffer) {
        AgentData_Field.AgentID = unpackUUID(byteBuffer)
        AgentData_Field.SessionID = unpackUUID(byteBuffer)
    }
}
